### ISSUE TITLE
Change names of FittingProblem classes

### ISSUE THREAD

**Issue by TOFarmer** [fitbenchmarking/fitbenchmarking#272]
**Description**
There are three identical classes all called `FittingProblem`, in the modules `parse_nist_data`, `parse_fitbenchmark_data` and `parse_sasview_data`.

**Suggested fix**
I would suggest giving these unique names (e.g. `NistFittingProblem`) to eliminate the risk of confusion/bad imports.

**Comment by AndrewLister-STFC** [comment]
I think it would be better to have 1 FittingProblem and 3 parsers which each return an instance of the FittingProblem

**Comment by TOFarmer** [comment]
I completely agree, I would definitely prefer this to be the case.  I thought that we had discussed this at the FitBenchmarking mini hackathon and there was some problem with this, but if not then we should definitely aim for a single `FittingProblem`.  Otherwise as a minimum we should rename the `FittingProblem` classes.

**Comment by AndrewLister-STFC** [comment]
No longer relevant after #264
